python3Packages.octave-kernel: init at 0.36.0

fred441a opened the pull request:
Allows people to use the popular GNU octave scientific programming language in jupyter. https://github.com/Calysto/octave_kernel


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

getchoo commented on pkgs/development/python-modules/octave-kernel/default.nix:
```diff
@@ -0,0 +1,53 @@
+    dependency-groups = { };
+  };
+
+  pyproject = true;
```
Bit of a nitpick, but usually the first attributes in a Python module expression are ordered like so

```nix
{
  pname = "...";
  version = "...";
  pyproject = true;

  disabled = pythonOlder "3.7";

  src = { };

  build-system = [ ];

  dependencies = [ ];

  meta = { };
}
```
This isn't a hardset rule or anything though, so feel free to ignore ;)

getchoo commented on pkgs/development/python-modules/octave-kernel/default.nix:
```diff
@@ -0,0 +1,53 @@
+    ];
+  };
+
+  passthru = {
+    dependency-groups = { };
+  };
```
```suggestion
```
Pretty sure we don't need this

getchoo reviewed (commented):
Also: welcome to Nixpkgs!

fred441a committed:
maintainers: add fred441a

fred441a committed:
python3Packages.octave-kernel: init at 0.36.0

Allows people to use the popular GNU octave scientific programming language in jupyter. https://github.com/Calysto/octave_kernel

fred441a commented:
> Also: welcome to Nixpkgs!

Thank you! Hope the new commit structure is better :)

getchoo reviewed (approved):
It looks like the `nixfmt` checks are [failing](https://github.com/NixOS/nixpkgs/actions/runs/12299947858/job/34326929412?pr=364327). You can fix this by running `nix-shell --command "nixfmt 'pkgs/development/python-modules/octave-kernel/default.nix'"` from the root of the repository

Otherwise LGTM. Feel free to ping in a couple days if there are no other reviews :)

fred441a commented:
@getchoo @natsukium @wegank Any more changes before this can be approved and merged?

FliegendeWurst commented on pkgs/development/python-modules/octave-kernel/default.nix:
```diff
@@ -0,0 +1,50 @@
+  ];
+
+  optional-dependencies = {
+    test = [
+      pytest
+      nbconvert
+    ];
```
Please add `nativeCheckInputs = [ pytestCheckHook nbconvert ];` to ensure the tests are run.

And maybe remove `optional-dependencies.test`. Unless there is a reason to keep it.

FliegendeWurst commented on pkgs/development/python-modules/octave-kernel/default.nix:
```diff
@@ -0,0 +1,50 @@
+  };
+
+  meta = with lib; {
+    description = "A Jupyter kernel for Octave.";
```
```suggestion
    description = "Jupyter kernel for Octave";
```

FliegendeWurst commented on pkgs/development/python-modules/octave-kernel/default.nix:
```diff
@@ -0,0 +1,50 @@
+  src = fetchFromGitHub {
+    owner = "Calysto";
+    repo = "octave_kernel";
+    rev = "refs/tags/v${version}";
```
```suggestion
    tag = "v${version}";
```
New parameter for fetching tags
